parsec/mca/device/device.c

```diff
@@ -873,16 +846,68 @@ static int cpu_weights(parsec_device_module_t* device, int nstreams)
         fp_ipc = 16;
         dp_ipc = 8;
     }
-#endif
-#endif  /* defined(__x86_64__) || defined(__i386__) */
+    else {
+        fp_ipc = 8;
+        dp_ipc = 4;
+    }
+#elif defined(__PPC64__)
+    fp_ipc = 16;
+    dp_ipc = 8;
+#endif  /* defined(__x86_64__) || defined(__i386__) || defined(__PPC64__) */
     free(cpu_flags);
 
+#elif defined(__APPLE__)
+    size_t len = sizeof(cpu_model);
+    size_t val = 0;
+    int rc = sysctlbyname("machdep.cpu.brand_string", cpu_model, &len, NULL, 0);
+    if( rc ) {
+        parsec_warning("CPU Name cannot be autodetected on this machine (OSX): %s", strerror(errno));
+        goto notfound;
+    }
+    /* is it an arm64 mac? */
+    rc = sysctlbyname("hw.optional.arm64", &val, &len, NULL, 0);
+    if( 0 == rc && val ) {
+        /* vector length */
+        fp_ipc = 16;
+        dp_ipc = 8;
+        // TODO: figure out a way to obtain cpu freq on arm64 macos
```

Comment:
`hw.cpufrequency` does not work ?

Reply:
not on arm cpus, there are multiple maximum cpu frequencies on arm chips (for e-cores and p-cores) and that cannot be queried through that simple interface, overall having this work was too much work for little usefulness